src/apps/lwaftr/conf_parser.lua

```diff
@@ -0,0 +1,292 @@
+   return addr_or_err
+end
+
+function Parser:parse_ipv4_range()
+   local range_begin, range_end
+   range_begin = self:parse_ipv4_as_uint32()
+   self:skip_whitespace()
+   if self:check('-') then
+      self:skip_whitespace()
+      range_end = self:parse_ipv4_as_uint32()
+   else
+      range_end = range_begin
+   end
+   if range_end < range_begin then
+      self:error('invalid IPv4 address range (end before begin)')
+   end
+   return { min=range_begin, max=range_end }
+end
+
+function Parser:parse_ipv4_range_list()
+   local ranges = {}
+   repeat
+      self:skip_whitespace()
+      table.insert(ranges, self:parse_ipv4_range())
+      self:skip_whitespace()
+   until not self:check(',')
+   return ranges
+end
+
+function Parser:parse_quoted_string(quote, escape)
+   local res = {}
+   escape = escape or '\\'
+   while not self:check(quote) do
+      local ch = self:next()
+      if ch == escape then ch = self:next() end
+      if not ch then self:error('EOF while reading quoted string') end
+      table.insert(res, ch)
+   end
+   return table.concat(res)
+end
+
+function Parser:parse_string()
+   local str
+   if self:check("'") then str = self:parse_quoted_string("'")
+   elseif self:check('"') then str = self:parse_quoted_string('"')
+   else str = self:take_while('[^%s,]') end
+   return str
+end
+
+function Parser:parse_file_name()
+   local str = self:parse_string()
+   if str == '' then self:error('file name is empty') end
+   -- Relative paths in conf files are relative to the location of the
+   -- conf file, not the current working directory.
+   if not str:match('^/') and self.name then
+      str = lib.dirname(self.name)..'/'..str
+   end
+   return str
+end
+
+function Parser:parse_boolean()
+   local tok = self:take_while('[a-zA-Z]')
+   if tok:lower() == 'true' then return true end
+   if tok:lower() == 'false' then return false end
+   self:error('expected "true" or "false", instead got "%s"', tok)
+end
+
+function Parser:parse_number()
+   local tok = self:take_while('[0-9.eExX]')
+   local num = tonumber(tok)
+   if not num then self:error('expected a number, instead got "%s"', tok) end
+   return num
+end
+
+function Parser:parse_positive_number()
+   local tok = self:take_while('[0-9.eExX]')
+   local num = tonumber(tok)
+   if not num then self:error('expected a number, instead got "%s"', tok) end
+   return num
+end
+
+function Parser:parse_positive_number()
+   local num = self:parse_number()
+   if num <= 0 then
+      self:error('expected a positive number, instead got %s',
+                 tostring(num))
+   end
+   return num
+end
+
+function Parser:parse_non_negative_number()
+   local num = self:parse_number()
+   if num < 0 then
+      self:error('expected a non-negative number, instead got %s',
+                 tostring(num))
+   end
+   return num
+end
+
+function Parser:parse_mtu()
+   -- FIXME: what's a maximum MTU?
+   return self:parse_uint(0,2^16)
```

Comment:
Maximum value of a MTU is something around 9000 bytes. 2^16 is the right definition since MTU is two bytes. If we would like to impose a logical limit I would set the maximum value in 10000.

Discusses maximum MTU values:

http://www.juniper.net/documentation/en_US/junos14.1/topics/reference/general/interface-security-physical-property-mtu-default-and-maximum-value.html